src/main/java/pl/damianszczepanik/jenkins/buildhistorymanager/BuildHistoryManager.java

```diff
@@ -48,25 +48,29 @@ public void perform(Job<?, ?> job) throws IOException, InterruptedException {
 
         Run<?, ?> run = job.getLastCompletedBuild();
         // for each completed build...
-        do {
+        while (run != null) {
```

Comment:
you can rebase the code so change from your other PR won't make noise here - please

Reply:
im trying to, give me a chance^^

Reply:
should be ok now